94.md

```diff
@@ -0,0 +1,49 @@
+NIP-94 - File Header
+======
+`draft` `optional` `author:frbitten` 
+
+The purpose of this NIP is to allow an organization and classification of shared files. So that relays can filter and organize in any way that is of interest.
+Also the goal is to create a base on the protocol for this bountsr "Filesharing App" (https://bountsr.org/p2p-filesharing/) to be implemented.
+
+Nostr event
+------------------
+This NIP specifies the use of the `1063` event type (parameterized replaceable event), having in `content` a description of the file content, and a list of tags described below:
+* `url` the url to download the file
+* `type` a string indicating the data type of the file. The MIME types format must be used (https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types)
+* `decrypt` (optional) If the file is encrypted, you must indicate in the first value the algorithm used to encrypt the file and in the second value the parameters that allow the decryption of the file
+* `hash` containing the SHA-256 hexencoded string of the file or the BlurHash string (for images). The third array value is the lowercase chosen algorithm name: "sha256" or "blurhash".
```

Comment:
Just ocurred to me that `h` instead of `hash` will make it possible to find by the tag, so to know if a file header already exists before effectively uploading a file.

```suggestion
* `h` containing the SHA-256 hexencoded string of the file or the BlurHash string (for images). The third array value is the lowercase chosen algorithm name: "sha256" or "blurhash".
```

Reply:
I think the best way would be to replace this with two tags: `h`, which contains the hex-encoded lowercase sha256 hash of the file, for search purposes (our own little IPFS), and `blurhash` in a separate tag, as it has just cosmetic purposes.

Reply:
Why not make it `h` so it is searchable? I think that, since users can write anything in the `sha256` field and not necessarily the correct hash, we will need relays to volunteer support for this extra requirement, if they want, and block events that have the wrong hashes, otherwise the system will not work.

If relays are already going through this tedious and intensive work of downloading the files and validating their hashes, they may as well support queries for the filter `#sha256`.